### PR TITLE
chore: single source of truth for options

### PR DESCRIPTION
**Pull request by techfg**
TL;DR - Achieve a single source of truth for options definition/usage and improve typings throughout the codebase.

The current code contains multiple points of definition & maintenance for the plugin options (e.g., we define them [here](https://github.com/vernak2539/astro-rehype-relative-markdown-links/blob/main/src/index.d.ts#L5) and again [here](https://github.com/vernak2539/astro-rehype-relative-markdown-links/blob/main/src/index.mjs#L33) and they are not exactly the same types).  The goal of this PR is to consolidate and get to a single source of truth whereby one definition is leveraged throughout the code base, it's strongly typed and has meaningful defaults (building on the work @vernak2539 did recently). 

In order to achieve this goal, a few other improvements were necessary and to make it as straightforward as possible to see the changes, there are four (4) commits in this PR to show the lineage.

1. [Commit ee81377](https://github.com/techfg/astro-rehype-relative-markdown-links/commit/ee8137792845e21a2f5dd25822cf55ded5121964)
    - There was a dependency on `unist-util-visit ^4.1.2` which is based on `unified^10.x` but also a dependency on `rehype 13.0.1` which is based on `unified ^11.x` so updated `unist-util-visit` to `^5.0.0` to unify (no pun intended lol)
    - Added `unified` as a dependency since its brought in any way and we now use it directly

2. [Commit 6c5331c](https://github.com/techfg/astro-rehype-relative-markdown-links/commit/6c5331c1a47ffb6a0422453811a051f029073968)
    - Plugin was missing a typing in `index.d.ts` for the default export `astroRehypeRelativeMarkdownLinks` in `index.mjs`
    - Also missing from docs

3. [Commit 6524bec](https://github.com/techfg/astro-rehype-relative-markdown-links/commit/6524bec8913ac48b0bc277cf6feafadafc6b63c7)
    - In order to accomplish the goal stated above, I needed to be able to specify `tsconfig` compiler options in order to properly generate the docs based on the usage of zod for the source of truth (see details below).  To meet this need, there were a couple options:
        1. Add a `tsconfig.json` to the project root
        2. Add a `typedoc` specific `tsconfig.json` (e.g., `typedoc-tsconfig.json`) using it only from `typedoc
    - I went with adding `tsconfig.json` to project root as it not only provides solution for generating proper documentation but also gives us full typescript support throughout entire code base
    - Since you didn't have `tsconfig.json` previously, possible that you intentionally didn't want it and if so, let me know and I can go with a typedoc specific version of `tsconfig.json` so its only used when generating docs and not throughout
    - Switched `typedoc` to use the default export (`src/index.mjs` instead of `src/index.d.ts`)
    - Added a `type-check` yarn script

4. [Commit 2aad2b3](https://github.com/techfg/astro-rehype-relative-markdown-links/commit/2aad2b3c595be91682dd563a2ba64a5a1ff20d59)
    - The zod schema is now the single source of truth for options - anywhere options are needed, it comes from the schema
    - Tests - I took a dependency on [esmock](https://github.com/iambumblehead/esmock) in order to spy on calls to the new [validationOptions](https://github.com/techfg/astro-rehype-relative-markdown-links/commit/2aad2b3c595be91682dd563a2ba64a5a1ff20d59#diff-d4200aac207dccf4aa6d49aec554d46d8c8464890ed95cde3233d1a8f77b5222R83) method to ensure that its called from `astroRehypeRelativeMarkdownLinks` in the refactored tests in [index.test.mjs](https://github.com/techfg/astro-rehype-relative-markdown-links/commit/2aad2b3c595be91682dd563a2ba64a5a1ff20d59#diff-f490656170ffac56d5a054c802ce770ae7e7e5f78a015f21225d9ed679f844ccR312) ([validateOptions](https://github.com/techfg/astro-rehype-relative-markdown-links/commit/2aad2b3c595be91682dd563a2ba64a5a1ff20d59#diff-3eb9d1d4c96898e659866e22155aa49da11a985720ecc51e0780e9e2d17ca366R8) has its own set of comprehensive tests).  Regarding esmock, see [my note](https://github.com/techfg/astro-rehype-relative-markdown-links/commit/2aad2b3c595be91682dd563a2ba64a5a1ff20d59#diff-f490656170ffac56d5a054c802ce770ae7e7e5f78a015f21225d9ed679f844ccR16) about it and why I had to introduce it.  I'll admit, we probably don't need these tests but I do think they are a good safeguard and we may need a module mocking solution at some point so we benefit both ways.  That said, if you'd prefer I remove these tests and eliminate the esmock depedency, just let me know.

Open to any thoughts/feedback!

### CODE REVIEW

**Comment by vernak2539** [comment]
Sorry it took so long to get this one done! Thanks for doing it, it looks great. Tbh, I've only done these thing when I have 5 minutes, so they're not always great. But, I really appreciate you jumping in and making things better!

**Comment by techfg** [comment]
lol, no need to apologize!  Great to see this one merged, ty!
